django_mongodb/base.py

```diff
@@ -0,0 +1,116 @@
+from django.core.exceptions import ImproperlyConfigured
+from django.db.backends.base.base import BaseDatabaseWrapper
+from django.db.backends.signals import connection_created
+from pymongo.collection import Collection
+from pymongo.mongo_client import MongoClient
+
+from . import dbapi as Database
+from .client import DatabaseClient
+from .creation import DatabaseCreation
+from .features import DatabaseFeatures
+from .introspection import DatabaseIntrospection
+from .operations import DatabaseOperations
+from .schema import DatabaseSchemaEditor
+
+
+class Cursor:
+    """A "nodb" cursor that does nothing except work on a context manager."""
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exception_type, exception_value, exception_traceback):
+        pass
+
+
+class DatabaseWrapper(BaseDatabaseWrapper):
+    data_types = {
+        "AutoField": "int",
+        "BigAutoField": "long",
+        "BinaryField": "binData",
+        "BooleanField": "bool",
+        "CharField": "string",
+        "DateField": "date",
+        "DateTimeField": "date",
+        "DecimalField": "decimal",
+        "DurationField": "long",
+        "FileField": "string",
+        "FilePathField": "string",
+        "FloatField": "double",
+        "IntegerField": "int",
+        "BigIntegerField": "long",
+        "GenericIPAddressField": "string",
+        "NullBooleanField": "bool",
+        "OneToOneField": "int",
+        "PositiveIntegerField": "long",
+        "PositiveSmallIntegerField": "int",
+        "SlugField": "string",
+        "SmallIntegerField": "int",
+        "TextField": "string",
+        "TimeField": "date",
+        "UUIDField": "string",
+    }
+
+    vendor = "mongodb"
+    Database = Database
+    SchemaEditorClass = DatabaseSchemaEditor
+    client_class = DatabaseClient
+    creation_class = DatabaseCreation
+    features_class = DatabaseFeatures
+    introspection_class = DatabaseIntrospection
+    ops_class = DatabaseOperations
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.connected = False
+        del self.connection
+
+    def get_collection(self, name, **kwargs):
+        return Collection(self.database, name, **kwargs)
+
+    def __getattr__(self, attr):
+        if attr in ["connection", "database"]:
+            assert not self.connected
```

Comment:
I am not getting the point. I think it should be 'if' instead of 'assert'. Why should it fail if I call 'self.connection' twice?

Reply:
`__getattr__()` is only called the first time when the attribute doesn't exist.